server/src/db/catalog.rs

```diff
@@ -321,6 +352,129 @@ impl SchemaProvider for Catalog {
+#[derive(Debug)]
+pub struct CatalogEmptyInput {
+    /// Memory registries used to track chunks.
+    pub memory_registries: Arc<MemoryRegistries>,
+
+    /// All of the metrics for this Db.
+    pub metrics: Arc<DbMetrics>,
+
+    /// Metrics domain
+    pub metrics_domain: metrics::Domain,
+}
+
+impl CatalogState for Catalog {
+    type EmptyInput = CatalogEmptyInput;
+
+    fn new_empty(data: Self::EmptyInput) -> Self {
+        Self::new(data.memory_registries, data.metrics, data.metrics_domain)
+    }
+
+    fn clone_or_keep(origin: &Arc<Self>) -> Arc<Self> {
+        // no copy semantics, just keep it
+        Arc::clone(origin)
+    }
+
+    fn add(
+        &self,
+        object_store: Arc<ObjectStore>,
+        server_id: ServerId,
+        db_name: &str,
+        path: DirsAndFileName,
+        metadata: ParquetMetaData,
+    ) -> parquet_file::catalog::Result<()> {
+        use parquet_file::catalog::{
+            CatalogStateFailure, PathParseFailed, SchemaReadFailed, StatisticsReadFailed,
+        };
+        // parse path
+        let storage = Storage::new(Arc::clone(&object_store), server_id, db_name.to_string());
+        let (partition_key, chunk_id, table_name) = storage
+            .parse_location(path.clone())
+            .context(PathParseFailed { path: path.clone() })?;
+
+        // change path represention
+        let path = object_store.path_from_dirs_and_filename(path);
+
+        // extract metadata
+        let schema = read_schema_from_parquet_metadata(&metadata)
+            .context(SchemaReadFailed { path: path.clone() })?;
+        let (table_summary, table_time_range) =
+            read_statistics_from_parquet_metadata(&metadata, &schema, &table_name)
+                .context(StatisticsReadFailed { path: path.clone() })?;
+
+        let chunk = {
+            let partition = self
+                .valid_partition(&partition_key)
+                .map_err(|e| Box::new(e) as _)
+                .context(CatalogStateFailure { path: path.clone() })?;
+            let partition = partition.read();
+
+            partition
+                .chunk(table_name, chunk_id)
+                .map_err(|e| Box::new(e) as _)
+                .context(CatalogStateFailure { path: path.clone() })?
+        };
+
+        // Create a parquet chunk for this chunk
+        let mut parquet_chunk = ParquetChunk::new(
+            partition_key,
+            chunk_id,
+            self.memory_registries.parquet.as_ref(),
+        );
+        parquet_chunk.add_table(table_summary, path, object_store, schema, table_time_range);
+
+        // Relock the chunk again (nothing else should have been able
+        // to modify the chunk state while we were moving it
+        let mut chunk = chunk.write();
+        let prev_chunk_state = (chunk.state().metric_label(), chunk.size());
+
+        // chunk state change
+        let parquet_chunk = Arc::clone(&Arc::new(parquet_chunk));
```

Comment:
I realize you moved this line the way it was in server/src/db.rs, but... do you know why this line is creating a new `Arc`, not holding onto it, and then immediately cloning it...? I don't know of a reason why this can't be:

```suggestion
        let parquet_chunk = Arc::new(parquet_chunk);
```

Reply:
done